### PR TITLE
Remove page url check during `history.pushState`

### LINKED ISSUE #9296: Strange behavior after Electron v1.3.5
* Electron version: >= 1.3.6
* Operating system: Windows 7 and 8, macOS 10.12.4 

### Expected behavior
See previous view
See previous view

### Actual behavior
Still on current view
Still on current view

### How to reproduce:
1) Enter your username/email then click "Next"
2) Click "Back"
___

I took this issue on large project and I tried to reproduce it and I got!
This issue works only after electron 1.3.6
On 1.3.5 and below or in in last versions of Chromium I don't get this behavior.

For reproducing this issue I used this simple application:
```javascript
const electron = require('electron');
const app = electron.app;
const BrowserWindow = electron.BrowserWindow;

app.on('ready', () => {
  const win = new BrowserWindow({
    'title'           : app.getName(),
    'width'           : 800,
    'height'          : 600
  });

  win.loadURL('https://login.live.com/');
  win.openDevTools();
  win.on('closed', app.quit);
});
```

### PR DESCRIPTION
**Pull request by tonyganch**
This should fix #9296.
Test repo: https://github.com/tonyganch/electron-quick-start/tree/history-regression

**TL;DR:** Current implementation of NavigationController does not allow using
`history.pushState()` if page url is not changed.
I think it worked by mistake in versions < 1.3.6 and got visible after fix 180a77e6.

**The issue:** live.com website uses `history.pushState()` and `history.back()` to navigate between different views of login form. Click on "Sign in" button calls `history.pushState({viewId: viewId}, '')` and click on "Back" button calls `history.back()`.
On `history.pushState()` NavigationController checks if the url of the page has changed and if not, then it doesn't push new state to history, meaning that `history.pushState()` called from js does absolutely nothing.
When the page then tries to call `history.back()`, NavigationController checks if the history is empty, sees that it is and does nothing (which makes sense).

Electron has its implementation of NavigationController which was introduced by this commit: 0143a4548869c1e8201ef3c3b6c2e098d5dd5140.
You can see that on line 19 we added the check if the page has the same url or a new one:
https://github.com/electron/electron/commit/0143a4548869c1e8201ef3c3b6c2e098d5dd5140#diff-35a86eec0f3148dfbe7621bd8f9bab7cR19
I don't see such check in previous code which makes me think that maybe it was introduced by mistake?

Commit after which everything "broke" is 180a77e6, but i honestly think that it just made the situation obvious and the things didn't work properly even before that.

### CODE REVIEW

**Comment by kevinsawicki** [comment]
> Test repo: https://github.com/tonyganch/electron-quick-start/tree/history-regression

@tonyganch would it be possible to get a test in this pull request that mirrors the behavior in your test repo?

**Comment by tonyganch** [comment]
@kevinsawicki, I added a test to `api-web-contents-spec.js` as it seemed to be the most appropriate place. I checked that the test fails without the fix and passes with it.

**Comment by tonyganch** [comment]
@kevinsawicki, i fixed a couple of linter errors, now all checks pass

**Comment by kevinsawicki** [comment]
Thanks for this @tonyganch 👍 🚀
